pkg/kubelet/config/common.go

```diff
@@ -58,9 +58,8 @@ func generatePodName(name string, nodeName types.NodeName) string {
 func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.NodeName) error {
 	if len(pod.UID) == 0 {
 		hasher := md5.New()
-		hash.DeepHashObject(hasher, pod)
-		// DeepHashObject resets the hash, so we should write the pod source
-		// information AFTER it.
+		fmt.Fprintf(hasher, "pod:%s", pod.Name)
+		fmt.Fprintf(hasher, "namespace:%s", pod.Namespace)
```

Comment:
Can we use structured logging here instead and collapse this into one logline? See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md